sympy/integrals/integrals.py

```diff
@@ -419,13 +419,10 @@ def doit(self, **hints):
             return S.Zero
 
         if isinstance(self.function, Sum):
-            for variable in self.variables:
-                if variable in self.function.limits[0]:
-                    break
-                else:
-                    _i = self
-                    _sum = self.function
-                    return _sum.func(_i.func(_sum.function, *_i.limits).doit(), *_sum.limits).doit()
+            if not any(v in self.function.limits[0] for v in self.variables):
+                _i = self
+                _sum = self.function
+                return _sum.func(_i.func(_sum.function, *_i.limits).doit(), *_sum.limits).doit()
 
```

Comment:
What should be done if an integration variable also appears in the limits of the sum, raise an exception?

Reply:
I don't think we should raise an exception. We can just output the same thing as was happening when this fix wasn't introduced, i.e output the same thing if that is ok?

Reply:
Well, if an integration variable also appears as an index or a limit, there is something wrong. Shouldn't we tell the user?

Reply:
@jksuom Sorry for the late reply, had classes today and difference in time zones.

So, I don't think we should give an exception due to the following example:

```
t = integrate(Sum(x**2, (x, 1, n)), n)
```

Now, for this, ```t.function == Sum(x**2, (x, 1, n)``` and clearly, ```t.variables == [n]```, ```n in t.function.limits[0] == True``` and also if we expand the sum, or use something like ```summation```, this integral can be easily computed despite the condition.

Reply:
I think that `Sum(x**2, (x, 1, n)` only makes sense for integral values of `n` though the result is a polynomial in `n`.

Reply:
@jksuom you are right. Now I see your point. If that is the case then it makes sense to raise an exception as if the integration variable appears in limits of a sum, then it makes no sense as limits of a sum is an integer(I think) and integration variable for a range is continuous. We should raise an exception.

Reply:
@jksuom I will add the exception in a format similar to below:

```
if any(v in self.function.limits[0] for v in self.variables):
    raise ValueError('Limit of the sum cannot be an integration variable.')
_i = self
_sum = self.function
return _sum.func(_i.func(_sum.function, *_i.limits).doit(), *_sum.limits).doit()
```